firmware/application/apps/ui_recon.cpp

```diff
@@ -354,16 +367,21 @@ void ReconView::focus() {
 }
 
 ReconView::~ReconView() {
-    // Save recon config.
+    recon_stop_recording();
+    delete record_view;
     recon_save_config_to_sd();
-
-    audio::output::stop();
+    if (field_mode.selected_index_value() != SPEC_MODULATION)
+        audio::output::stop();
+    receiver_model.set_modulation(ReceiverModel::Mode::WidebandFMAudio);
     receiver_model.disable();
     baseband::shutdown();
 }
 
 ReconView::ReconView(NavigationView& nav)
     : nav_{nav} {
+    chrono_start = chTimeNow();
+
+    record_view = new RecordView({0, 0, 30 * 8, 1 * 16}, u"AUTO_AUDIO_", u"AUDIO", RecordView::FileType::WAV, 4096, 4);
```

Comment:
You should use a unique_ptr member so ensure you don't accidentally leak.

Reply:
basically the same usage pattern, but instead of calling `new`, use `std::make_unique<RecordView>(<args>)`

Reply:
this would be `record_view = std::make_unique<RecordView>({0, 0, 30 * 8, 1 * 16}, u"AUTO_AUDIO_", u"AUDIO", RecordView::FileType::WAV, 4096, 4);`